### PR TITLE
feat: Use semver versioning for Golang API package

### PR DESCRIPTION
**Pull request by victorcolombo**

### CODE REVIEW

File: .github/workflows/golang-package-tag.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+on:
+  push:
+    branches: [main]
+  tags-ignore:
+    - '*/v/*'
```
Comment (h4ck3rk3y) [inline .github/workflows/golang-package-tag.yml]: lets filter to just tags here

Comment (h4ck3rk3y) [reply .github/workflows/golang-package-tag.yml]: [0-9]+.[0-9]+.[0-9]+

File: .github/workflows/golang-package-tag.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  tag_golang_package:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: get-version
+        run: export CURRENT_VERSION="$(cat version.txt)"
+      - name: tag-golang-package
```
Comment (h4ck3rk3y) [inline .github/workflows/golang-package-tag.yml]: dont need yhpens

File: .github/workflows/golang-package-tag.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        uses: pkgdeps/git-tag-action@v2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (h4ck3rk3y) [inline .github/workflows/golang-package-tag.yml:16]: some of these should be deafults no?

File: .github/workflows/golang-package-tag.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+          github_repo: ${{ github.repository }}
+          version: ${{ env.CURRENT_VERSION }}
+          git_commit_sha: ${{ github.sha }}
+          git_tag_prefix: "golang/v"
```
Comment (h4ck3rk3y) [inline .github/workflows/golang-package-tag.yml]: "api/golang" ?

File: .github/workflows/golang-package-tag.yml (hunk 3 above)
Comment (victorcolombo) [reply .github/workflows/golang-package-tag.yml:16]: They are not, according to their docs

**Review by h4ck3rk3y — APPROVED** [review]
God speed

**Pull request merged**